### DISCUSSION TITLE
Edge Functions - support Dart

### DISCUSSION THREAD

**Discussion by ajonno** [supabase/supabase#7718] [Feature Requests]
Any plans to support Edge Functions authored using Dart ?

**Comment by dshukertjr** [comment]
Hi @ajonno,

Actually, Edge Functions is already supported on both [supabase-dart](https://pub.dev/packages/supabase) and [supabase-flutter](https://pub.dev/packages/supabase_flutter). 

You can find the docs on how to use it on our official Dart documentation [here](https://supabase.com/docs/reference/dart/invoke)!

**Reply by ajonno** [reply]
Sorry just to clarify, I meant can the Edge Function itself be written using Dart (not Typescript)?

**Reply by dshukertjr** [reply]
@ajonno Ah, I see! At the moment, we use Deno deploy for our edge functions, so there are no plans to support Dart, or any other languages I believe. With that being said, Supabase is always open for feedbacks from the community, so if enough people want it, I'm sure we won't ignore that!

**Comment by tananga** [comment]
@dshukertjr  @ajonno  This is a must have for Flutter developers. Why I need to learn another programming language just to wrote an edge function? Please this feature would help a lot. Firebase supports Python, Javascript, and few other languages but  they dont support dart. As a Flutter developer I dont wanna learn other languages and libraries which Im not gonna use in the future.

**Reply by illia-romanenko** [reply]
I'm pretty sure that Firebase will release cloud functions using Dart. That would be interesting then.

**Comment by lewinpauli** [comment]
@dshukertjr I'm reviving this thread because there have been some interesting developments worth flagging!

@illia-romanenko called it back in December 2022 — Firebase has now officially released Dart support for Cloud Functions: https://www.youtube.com/watch?v=0P17edjvTSw. This shows there's real demand in the Flutter community for server-side Dart, and that it's technically viable.

And the demand is only going to grow: according to Apptopia (via the [official Flutter blog, Dec 2024](https://developers.googleblog.com/en/celebrating-flutters-production-era/)), Flutter now powers **nearly 30% of all free iOS apps** and over **1 million apps on the Play Store**, with over **1 million monthly active developers**. A huge portion of those developers are already using Supabase as their backend — and having to switch to TypeScript just to write an edge function is a real pain point.

For Supabase specifically, the docs already reference [dart_edge](https://supabase.com/docs/guides/functions/dart-edge) as a way to write Edge Functions in Dart — which suggests this has been on the team's radar before. Unfortunately, the package is currently unmaintained and no longer builds, leaving developers with no working path today.

One potential technical approach would be compiling Dart to WebAssembly (WASM) and running it inside Deno. However, the current WASM approach is quite hacky — making outbound HTTP calls from within a WASM module in Deno is not straightforward, making it impractical for real-world use cases like calling external APIs or Supabase itself.

A proper first-class implementation would be a game changer for the Flutter community. Would love to know if this is something the team is considering revisiting!
